src/synapse/api.py

```diff
@@ -223,3 +261,133 @@ def get_version() -> str:
+    rate_limit_url = (
+        f"{SYNAPSE_URL}/_synapse/admin/v1/users/"
+        f"@{user.username}:{server_name}/override_ratelimit"
+    )
+    try:
+        authorization_token = f"Bearer {access_token}"
+        headers = {"Authorization": authorization_token}
+        res = requests.delete(rate_limit_url, headers=headers, timeout=5)
+        res.raise_for_status()
+    except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as exc:
+        logger.exception("Failed to connect to %s: %r", rate_limit_url, exc)
+        raise NetworkError(f"Failed to connect to {rate_limit_url}.") from exc
+    except requests.exceptions.HTTPError as exc:
+        logger.exception("HTTP error from %s: %r", rate_limit_url, exc)
+        raise NetworkError(f"HTTP error from {rate_limit_url}.") from exc
+
+
+def get_room_id(
+    room_name: str,
+    access_token: str,
+) -> str:
+    """Get version.
+
+    Args:
+        room_name: room name.
+        access_token: access token to be used.
+
+    Returns:
+        The room id.
+
+    Raises:
+        GetRoomIDError: if there was an error while getting room id.
+    """
+    authorization_token = f"Bearer {access_token}"
+    headers = {"Authorization": authorization_token}
+    res = _send_get_request(LIST_ROOMS_URL, headers=headers)
+    try:
+        rooms = res.json()["rooms"]
+        for room in rooms:
+            if room["name"].upper() == room_name.upper():
+                return room["room_id"]
+    except (requests.exceptions.JSONDecodeError, TypeError, KeyError) as exc:
+        logger.exception("Failed to decode rooms: %r. Received: %s", exc, res.text)
+        raise GetRoomIDError(str(exc)) from exc
+
+    raise GetRoomIDError(f"Room {room_name} not found.")
+
+
+def deactivate_user(
+    user: User,
+    server: str,
+    access_token: str,
+) -> None:
+    """Deactivate user.
+
+    Args:
+        user: user to be deactivated.
+        server: to be used to create the user id.
+        access_token: access token to be used.
+
+    Raises:
+        NetworkError: if there was an error registering the user.
+    """
+    authorization_token = f"Bearer {access_token}"
+    headers = {"Authorization": authorization_token}
+    data = {
+        "erase": True,
+    }
+    try:
+        # @user:server.com
+        user_id = f"@{user.username}:{server}"
+        res = requests.post(
+            f"{DEACTIVATE_ACCOUNT_URL}/{user_id}", headers=headers, json=data, timeout=5
+        )
+        res.raise_for_status()
+    except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as exc:
+        logger.exception("Failed to connect to %s: %r", DEACTIVATE_ACCOUNT_URL, exc)
+        raise NetworkError(f"Failed to connect to {DEACTIVATE_ACCOUNT_URL}.") from exc
+    except requests.exceptions.HTTPError as exc:
+        logger.exception("HTTP error from %s: %r", DEACTIVATE_ACCOUNT_URL, exc)
+        raise NetworkError(f"HTTP error from {DEACTIVATE_ACCOUNT_URL}.") from exc
```

Comment:
This is repeated several time, I'm wondering if you couldn't write a decorator to help there, or make a generic method that wraps requests.request (https://requests.readthedocs.io/en/latest/api/#requests.request)

Reply:
Excellent suggestion, changed!